src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+        self.assertIsInstance(megafield_metadata.x_scan_to_acq_delay, int)
+        self.assertIsInstance(megafield_metadata.x_cell_size, int)
+        self.assertIsInstance(megafield_metadata.x_eff_cell_size, int)
+        self.assertIsInstance(megafield_metadata.x_scan_gain, float)
+        self.assertIsInstance(megafield_metadata.x_scan_offset, float)
+        self.assertIsInstance(megafield_metadata.x_descan_offset, int)
+        self.assertIsInstance(megafield_metadata.y_cell_size, int)
+        self.assertIsInstance(megafield_metadata.y_eff_cell_size, int)
+        self.assertIsInstance(megafield_metadata.y_scan_gain, float)
+        self.assertIsInstance(megafield_metadata.y_scan_offset, float)
+        self.assertIsInstance(megafield_metadata.y_descan_offset, int)
+        self.assertIsInstance(megafield_metadata.y_prescan_lines, int)
+        self.assertIsInstance(megafield_metadata.x_scan_delay, int)
+        self.assertIsInstance(megafield_metadata.scan_rotation, float)
+        self.assertIsInstance(megafield_metadata.descan_rotation, float)
+
+        # Test descan setpoints
+        self.assertIsInstance(megafield_metadata.x_descan_setpoints, list)
+        self.assertIsInstance(megafield_metadata.y_descan_setpoints, list)
+        for x_setpoint, y_setpoint in zip(megafield_metadata.x_descan_setpoints, megafield_metadata.y_descan_setpoints):
+            self.assertIsInstance(x_setpoint, int)
+            self.assertIsInstance(y_setpoint, int)
+
+        # Test cell_parameters
+        self.MPPC.cellTranslation.value = \
+            tuple(tuple((10 + j, 20 + j) for j in range(i, i + self.MPPC.shape[0]))
+                  for i in range(0, self.MPPC.shape[1] * self.MPPC.shape[0], self.MPPC.shape[0]))
+
+        self.MPPC.cellDarkOffset.value = \
+            tuple(tuple(j for j in range(i, i + self.MPPC.shape[0]))
+                  for i in range(0, self.MPPC.shape[1] * self.MPPC.shape[0], self.MPPC.shape[0]))
+
+        self.MPPC.cellDigitalGain.value = \
+            tuple(tuple(float(j) for j in range(i, i + self.MPPC.shape[0]))
+                  for i in range(0, self.MPPC.shape[1] * self.MPPC.shape[0], self.MPPC.shape[0]))
+
+        megafield_metadata = self.MPPC._assembleMegafieldMetadata()
+        self.assertEqual(len(megafield_metadata.cell_parameters), self.MPPC.shape[0] * self.MPPC.shape[1])
+
+        for cell_number, individual_cell in enumerate(megafield_metadata.cell_parameters):
+            self.assertEqual(individual_cell.digital_gain, cell_number)
+            self.assertEqual(individual_cell.x_eff_orig, 10 + cell_number)
+            self.assertEqual(individual_cell.y_eff_orig, 20 + cell_number)
+            self.assertIsInstance(individual_cell.digital_gain, float)
+            self.assertIsInstance(individual_cell.x_eff_orig, int)
+            self.assertIsInstance(individual_cell.y_eff_orig, int)
+
+        # Test multiple stage positions and check if both ints and floats works
```

Comment:
Here I think one would be fine and would also do it at start of test same way as suggested for the cell parameters.  If you test for a float, should the int not also work?

Reply:
The test for both int and float is to test if the API object can still handle both (in some cases it can handle a float but not an int).

Reply:
About starting the test with changing the values I have responded in the conversation above.